src/atomWithHash.ts

```diff
@@ -26,6 +26,12 @@ export function atomWithHash<Value>(
   },
 ): WritableAtom<Value, [SetStateActionWithReset<Value>], void> {
   const serialize = options?.serialize || JSON.stringify;
+  let hashValueInURL;
+  if (global.window) {
+    hashValueInURL = new URLSearchParams(window.location.hash.slice(1)).get(
+      key,
+    );
+  }
```

Comment:
Maybe just out of preference, but can we do this?
```suggestion
  let initialValue: string | null = null;
  if (typeof window !== 'undefined') {
    initialValue = new URLSearchParams(window.location.hash.slice(1)).get(
      key,
    );
  }
```

Reply:
Don't think we can use `let initialValue` here since we already use that name for one of the props. I adjusted the inner variable to `initialValueInHash` but I'm open to other suggestions!
```js
export function atomWithHash<Value>(
  key: string,
  initialValue: Value,
  options?: {
    serialize?: (val: Value) => string;
    deserialize?: (str: string) => Value;
    subscribe?: (callback: () => void) => () => void;
    setHash?: 'default' | 'replaceState' | ((searchParams: string) => void);
  },
): WritableAtom<Value, [SetStateActionWithReset<Value>], void>
```

Reply:
Fixed `global.window` to `typeof window...` in https://github.com/jotaijs/jotai-location/pull/31/commits/80c21f93cc1e54325b63f0ebd98082ae1f1e7510

Reply:
> Don't think we can use `let initialValue` here

Ah, nice catch. Yeah, I'm okay with your suggestion.